buildlib/az-distro-release.yml

```diff
@@ -9,43 +13,71 @@ jobs:
       ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
         POSTFIX: ucx-pr$(System.PullRequest.PullRequestNumber)
 
-    pool:
-      name: MLNX
-      demands:
-        - harbor_registry -equals yes
+    ${{ if eq(parameters.name, 'x86_64') }}:
+      strategy:
+        matrix:
+          centos7_cuda11_x86_64:
+            build_container: centos7_cuda11_x86_64
+            artifact_name: $(POSTFIX)-centos7-mofed5-cuda11-x86_64.tar.bz2
```

Comment:
can we do something like
```
build_container: centos7_cuda11_${{ parameters.name }}
artifact_name: $(POSTFIX)-centos7-mofed5-cuda11-${{ parameters.name }}.tar.bz2
```

in order to avoid `${{ if eq(parameters.name, 'x86_64') }}:`

Reply:
No, it will cause duplicate key names:
```
build_container: centos7_cuda11_${{ parameters.name }}   # x86_64
build_container: centos7_cuda11_${{ parameters.name }}   # aarch64
```

Reply:
It will fail on Azure YAML schema validation.

Reply:
why? container name cannot be parametrized?

Reply:
Azure limitations. Here I tested it:
https://github.com/openucx/ucx/pull/9016
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=61588

Reply:
![image](https://user-images.githubusercontent.com/2255631/232315017-6f62c1d9-8467-41ab-af70-6f18eaa17718.png)


`centos8_cuda11_${{ parameters.name }}:` is really listed twice in the file.
Can you try to remove this duplication?

Reply:
Right, those dups might be removed. Yet, we have a different set of containers per OS architecture.
We can either reduce the support matrix or separate container lists by `if`.

Reply:
maybe first list all common containers, and then list specific containers under "if" condition?

Reply:
This will work!

Reply:
Addressed by the latest commit + changed the key name to better reflect expected values.